### DISCUSSION TITLE
oo/this-super

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#128] [Announcements]
# oo/this-super

Java程序员进阶之路，小白的零基础Java教程，Java中this和super的用法总结

https://tobebetterjavaer.com/oo/this-super.html

**Comment by DylanLJC** [comment]
void method1(ThisAsParam p){ //这里的ThisAsParam 是一个数据类型吗？这里是什么含义？

**Reply by itwanger** [reply]
搜一下这个关键字，你就知道，这是一个自定义类

**Comment by Shawson755** [comment]
每当创建一个子类对象的时候，也会隐式的创建父类对象，由 super 关键字引用。为什么我在其他地方看到说的是不会隐式的创建父类对象呢？所以到底谁对谁错啊？

**Reply by javabcde** [reply]
他应该是写错了 但是会调用构造方法 父类引用子类对象 new 反编译之后 也只有一个new 说明只有一个对象被创建了

**Reply by javabcde** [reply]
帮你找到一个 R大的回答 希望可以帮到你 https://www.zhihu.com/question/41810504/answer/117132716
